qdrant-landing/content/articles/why-rust.md

```diff
@@ -0,0 +1,48 @@
+
+Python, being just a bit younger than Java, is ubiquitous in ML projects, mostly owing to its tooling (notably jupyter notebooks), being easy to learn and integration in most ML stacks. It doesn’t have a traditional garbage collector, opting for ubiquitous reference counting instead, which somewhat helps memory consumption. With that said, unless you only use it as glue code over high-perf modules, you may find yourself waiting for results. Also getting complex python services to perform stably under load is a serious technical challenge.
+
+So Andrey looked around at what younger languages would fit the challenge. After some searching, two contenders emerged: Go and Rust. Knowing neither, Andrey consulted the docs, and found hinself intrigued by Rust with its promise of Systems Programming without pervasive memory unsafety.
+
+This early decision has been validated time and again. When first learning Rust, the compiler’s error messages are very helpful (and have only improved in the meantime). It’s easy to keep memory profile low when one doesn’t have to wrestle a garbage collector and has complete control over stack and heap. Apart from the much advertised memory safety, many footguns one can run into when writing C++ have been meticulously designed out. And it’s much easier to parallelize a task if one doesn’t have to fear data races.
+
+With Qdrant written in Rust, we can offer cloud services that don’t keep us awake at night, thanks to Rust’s famed robustness. A current qdrant docker container comes in at just a bit over 50MB — try that for size. As for performance… have some [benchmarks](https://qdrant.tech/benchmarks).
+
+| Engine   | setup_name             | Dataset               | Upload Time (s) | Upload + Index Time (s) | P95 (s) | RPS     | P99 (s) | Latency (s) | Precision | engine_params                    |
+|----------|------------------------|-----------------------|-----------------|-------------------------|---------|---------|---------|------------–|-----------|----------------------------------|
+| qdrant   | qdrant-rps-m-64-ef-512 | deep-image-96-angular | 845.78          | 8959.44                 | 0.055   | 1541.86 | 0.063   | 0.024       | 0.96      | {"search_params":{"hnsw_ef":64}} |
+| weaviate | weaviate-m-16-ef-128   | deep-image-96-angular | 8922.25         | 8922.25                 | 0.35    |  507.33 | 0.41    | 0.19        | 0.94      | {"vectorIndexConfig":{"ef":256}} |
+
+(Please note that the performance difference is not only down to the languages used, the implementations differ in algorithms and design tradeoffs. As always, use your own best judgement.)
```

Comment:
let's remove, maybe? I don't think the main factor is the language in this particular case